src/classes/REL_RelationshipsViewer_CTRL.cls

```diff
@@ -34,92 +34,145 @@
+        }
+
+        if (isInitialLoad) {
             //base node
             Node basenode = new Node();
-      
-            Contact c = [select LastName, FirstName, id, Title, Account.Name from Contact where id = :cid];
-            basenode.LastName = c.LastName;
-            basenode.FirstName = c.FirstName;
-            basenode.Title = c.Title;
-            basenode.AccountName = c.Account.Name;
+
+            Contact c = [SELECT
+                            LastName,
+                            FirstName,
+                            Id,
+                            Title,
+                            Account.Name
+                        FROM Contact
+                        WHERE Id = :contactId];
+
+            basenode.lastName = c.LastName;
+            basenode.firstName = c.FirstName;
+            basenode.title = c.Title;
+            basenode.accountName = c.Account.Name;
             basenode.id = c.id;
-            returnList.add(basenode);
-        }        
-        else{
-            relationshipList.clear();
-            returnList.clear();
-        }        
-        
-        relationshipList = [select id, npe4__RelatedContact__r.LastName, toLabel(npe4__Type__c), npe4__Status__c, npe4__RelatedContact__r.FirstName, 
-                                    npe4__RelatedContact__r.id, npe4__RelatedContact__r.Account.Name, npe4__RelatedContact__r.Title,
-                                    npe4__ReciprocalRelationship__r.id 
-                                    from npe4__Relationship__c 
-                                    where npe4__Contact__c = :cid and id NOT IN :renderedRelationships 
-                                    and npe4__ReciprocalRelationship__r.id NOT IN :renderedRelationships];
-
-        for (npe4__Relationship__c r : relationshipList){
-            Node n = new Node();             
-            n.LastName = r.npe4__RelatedContact__r.LastName;
+            nodes.add(basenode);
+        } else {
+            relationships.clear();
+            nodes.clear();
+        }
+
+        relationships = [SELECT
+                                Id,
+                                npe4__RelatedContact__r.LastName,
+                                toLabel(npe4__Type__c),
+                                npe4__Status__c,
+                                npe4__RelatedContact__r.FirstName,
+                                npe4__RelatedContact__r.id,
+                                npe4__RelatedContact__r.Account.Name,
+                                npe4__RelatedContact__r.Title,
+                                npe4__ReciprocalRelationship__r.id
+                        FROM    npe4__Relationship__c
+                        WHERE   npe4__Contact__c = :contactId AND
+                                id NOT IN :renderedRelationships AND
+                                npe4__ReciprocalRelationship__r.id NOT IN :renderedRelationships];
+
+        for (npe4__Relationship__c r : relationships) {
```

Comment:
Avoid single character variable names. I know the style guide allows for them in small for loops, but I'd rather see them avoid when the letter has some meaning.

Reply:
Agreed